bundles/org.openhab.ui.dashboard/src/main/java/org/openhab/ui/dashboard/internal/DashboardService.java

```diff
@@ -202,4 +227,25 @@ private void addTilesForExternalServices(Map<String, Object> properties) {
             }
         }
     }
+
+    /**
+     * Returns the localized text for the given key. When the key is 'locale' it returns the locale. If no locale or
+     * fall-back would be present it returns the key.
+     *
+     * @param key key to get locale from
+     * @param locale known locale
+     * @return localized text for the key
+     */
+
+    private String getLocalizedText(String key, Locale locale) {
+        Locale useLocale = locale == null ? localeProvider.getLocale() : locale;
+
+        if ("locale".equals(key)) {
+            return bundleContext.getBundle()
+                    .getEntry("ESH-INF/i18n/dashboard_" + useLocale.getLanguage() + ".properties") != null
+                            ? useLocale.getLanguage() : "en";
```

Comment:
Just to be clear: this will cover missing translation files AND missing single translation strings and fall back to english? If so please update the comment above.

Reply:
Comment added